clients/hadoopfs/src/main/java/io/lakefs/LakeFSFileSystem.java

```diff
@@ -218,4 +242,57 @@ public boolean seekToNewSource(long l) throws IOException {
             return false;
         }
     }
+
+    private Location pathToLocation(Path path) {
+        if (!path.isAbsolute()) {
+            path = new Path(this.workingDirectory, path);
+        }
+
+        URI uri = path.toUri();
+        if (uri.getScheme() != null && uri.getPath().isEmpty()) {
+            return null;
+        }
+
+        Location loc = new Location();
+        loc.setRepository(uri.getHost());
+        String s = uri.getPath().substring(1);
```

Comment:
What char are you cleaning here? can you document the uri format with an example uri? you can add the example in the method docs and it will help to understand the path parsing.